src/util/moneystr.cpp

```diff
@@ -13,9 +13,13 @@ std::string FormatMoney(const CAmount& n)
 {
     // Note: not using straight sprintf here because we do NOT want
     // localized number formatting.
-    int64_t n_abs = (n > 0 ? n : -n);
-    int64_t quotient = n_abs/COIN;
-    int64_t remainder = n_abs%COIN;
+    static_assert(COIN > 0);
```

Comment:
Shouldn't this be `> 1` not `> 0`? If COIN=1, the same undefined behavior is back.

Reply:
Now asserting `COIN > 1` :)